config/database.yml

```diff
@@ -17,6 +17,8 @@
 default: &default
   adapter: postgresql
   encoding: unicode
+  collation: C.UTF-8
```

Comment:
The Postgres default is `en_US.UTF-8`. `C.UTF-8` means the DB uses locale-agnostic (doesn't mean almighty) settings and `en_US.UTF-8` means sorting etc. tailored for US English, that I believe affects matching of accented characters and such. However I don't have access to DB at this moment so let me test it out later for more concrete answer.

Reply:
As far as I tested:

- **Collation** (which governs sorting order): `C.UTF-8` sorts every string by Unicode code point, which is the dumbest and most straightforward algorithm. `en_US.UTF-8` seems to fold casing and remove accents (as much as possible) from alphabets before sorting by English alphabetical order, but does nothing fancy for scripts other than the Latin alphabet.
- **Ctype** (governs case conversion and numeral format etc.): `C.UTF-8` seems to use Unicode's default settings while `en_US.UTF-8` uses (obviously) that of American English, but they are practically identical in the scope of text search.

Those settings mainly affect index efficiency, which means if you perform search in `en_US` settings against a DB of same `en_US` settings, it is expected to return results faster, but otherwise it'd cost extra processing time. This fact may lead to two kinds of opinions:

1. A matching or search query needs to be (eventually) operated under different collation rules corresponding to user locale. So we should better let the creator choose which locale they want, according to what language they expect most used on their instance, instead of automatically set up by the system. As such, we should not specify the initial locale in the config (no addition).
2. A matching or search query needs to be (eventually) operated under different collation rules corresponding to user locale, and DB collation is hardly meaningful in relevance search either way. In this case, it is better to set as `C.UTF-8` to avoid [additional overhead brought by locale-specific smart sorting](https://qiita.com/fujii_masao/items/2a715fb5a3f718d22ab4)[^1].

[^1]: Sorry for a Japanese article, but you can only read codes and tables.

Reply:
I think we should hold off on changing this file for now. As you said, there's barely any difference between `C.UTF-8` and `en_US.UTF-8` and the latter runs on our prod DB for DEV and potentially also our cloud-hosted setup. I'd rather revisit the whole topic of databases separately in one point instead of making this one change in a general cleanup PR.